diff --git a/src/idpyoidc/storage/jsonfile.py b/src/idpyoidc/storage/jsonfile.py
new file mode 100644
index 0000000..52cc516
--- /dev/null
+++ b/src/idpyoidc/storage/jsonfile.py
@@ -0,0 +1,38 @@
+import os
+import logging
+import json
+
+from filelock import FileLock
+
+logger = logging.getLogger(__name__)
+
+
+class JsonReadOnlyFile(object):
+
+    def __init__(self, file_name):
+        self.file_name = file_name
+
+        if not os.path.exists(file_name):
+            fp = open(file_name, "x")
+            fp.close()
+
+    def _read_info(self, fname):
+        if os.path.isfile(fname):
+            try:
+                lock = FileLock(f"{fname}.lock")
+                with lock:
+                    fp = open(fname, "r")
+                    info = json.loads(fp.read())
+                lock.release()
+                return info
+            except Exception as err:
+                logger.error(err)
+                raise
+        else:
+            _msg = f"No such file: '{fname}'"
+            logger.error(_msg)
+        return None
+
+    def __call__(self):
+        return self._read_info(self.file_name)
+
