diff --git a/src/fedservice/defaults.py b/src/fedservice/defaults.py
index bf33153..8a78a72 100644
--- a/src/fedservice/defaults.py
+++ b/src/fedservice/defaults.py
@@ -46,15 +46,15 @@ SERVICES = {
         "kwargs": {}
     },
     "trust_mark_status": {
-        "class": 'fedservice.entity.client.trust_mark_status.TrustMarkStatus',
+        "class": 'fedservice.trust_mark_entity.client.trust_mark_status.TrustMarkStatus',
         "kwargs": {}
     },
     "trust_mark_list": {
-        "class": 'fedservice.entity.client.trust_mark_list.TrustMarkList',
+        "class": 'fedservice.trust_mark_entity.client.trust_mark_list.TrustMarkList',
         "kwargs": {}
     },
     "trust_mark": {
-        "class": 'fedservice.entity.client.trust_mark.TrustMark',
+        "class": 'fedservice.trust_mark_entity.client.trust_mark.TrustMark',
         "kwargs": {}
     },
     "resolve": {
@@ -104,17 +104,17 @@ FEDERATION_ENDPOINTS = {
     },
     "trust_mark_status": {
         "path": "trust_mark_status",
-        "class": 'fedservice.entity.server.trust_mark_status.TrustMarkStatus',
+        "class": 'fedservice.trust_mark_entity.server.trust_mark_status.TrustMarkStatus',
         "kwargs": {}
     },
     "trust_mark": {
         "path": "trust_mark",
-        "class": 'fedservice.entity.server.trust_mark.TrustMark',
+        "class": 'fedservice.trust_mark_entity.server.trust_mark.TrustMark',
         "kwargs": {}
     },
     "trust_mark_list": {
         "path": "trust_mark_list",
-        "class": 'fedservice.entity.server.trust_mark_list.TrustMarkList',
+        "class": 'fedservice.trust_mark_entity.server.trust_mark_list.TrustMarkList',
         "kwargs": {}
     },
     "metadata_verification": {
